Documentation/developers/debugging.rst

```diff
@@ -193,7 +193,186 @@ If you use CLion, you can debug Polycube with the following steps:
+A port defined on a shadow cube is also visible from the network namespace:
+
+- the user can decide to configure the ports using Linux (e.g. ifconfig or the ip command) or polycubectl; for example: ``polycubectl <cubename> ports <PortName> set ip=<IpAddress>`` it is the same as ``ip netns exec pcn-<cubename> ifconfig <PortName> <IpAddress>``.
+- the developer can let Linux handle some traffic by sending it to the namespace (e.g. ARP, ICMP, but in general all those protocols able to be managed by a tool running inside the namespace);
+
+So, let's create the r1 router:
+
+
+::
+
+	polycubectl router add r1 shadow=true  
+	
+
+Now let's add to the router r1 a port for each veth (to_vethX) with an appropriate IP address:
+
+
+::
+
+	for i in `seq 1 2`;
+	do
+		polycubectl router r1 ports add to_veth${i} ip=10.0.${i}.254/24
+		polycubectl router r1 ports to_veth${i} set peer=veth${i}
+	done
+
+
+
+As for the span mode, when it is activated it shows all the traffic seen by the service also at the namespace.
+
+::
+
+	polycubectl router r1 set span=true
+
+
+To see what has just been created you can use these following commands:
+
+* ``sudo ip netns``: from the output you can see three namespaces created, ns1, ns2 and in particular pcn-r1 which is created specifically for the shadow cube;
+* ``sudo ip netns exec ns1 ip a`` (equivalent for ns2): from the output it is possible the network interfaces of the namespace ns1, in this case ``veth1``, with the assigned ip address and other information;
+* ``ip a``: as before but in this case we see the network interfaces of the default namespace, in particular ``r1-to_veth1`` and ``r1-to_veth2``
+
+
+
+
+
+If you want to see the topology, just use ``polycubectl topology``, it should show something like this:
+
+::
+
+	loglevel  name  service-name  shadow  span  type    name (ports)      peer (ports)
+	INFO      r1    router        true    true  TC      ns_port_to_veth1  r1-to_veth1
+                                                     	    ns_port_to_veth2  r1-to_veth2
+                                                      	    to_veth1          veth1
+                                                      	    to_veth2          veth2
+
+Or, if you want to see information about r1 ports, just use ``polycubectl r1 ports show`` and it should show something like this:
+
+::
+
+  ip             mac                name      peer   status  uuid
+  10.0.1.254/24  ba:e7:da:ec:49:17  to_veth1  veth1  UP      3e92bd2f-567d-4adc-b7c1-820eb9cca9a7
+  10.0.2.254/24  46:f3:b4:48:2e:7f  to_veth2  veth2  UP      b51d21f3-366a-4f39-b3af-58e07cf62c44
+
+
+
+tcpdump/Wireshark
+*****************
+
+Now, let's use Wireshark/tcpdump on the interface we want.
+
+For tcpdump we only need to do ``tcpdump -ni <interface-name>``. For example, if we want to see the traffic passing through router r1's ``to_veth1`` and we only want ICMP traffic, we need to give this command:
+
+::
+
+	sudo tcpdump icmp -ni r1-to_veth1
```

Comment:
I believe in the new example we should use "ip netns exec...." in order to start tcpdump in the proper (shadow) namespace.